signal_scanner_bot/env.py

```diff
@@ -13,8 +14,43 @@
 class _State:
     """Class for holding global state across threads/tasks"""
 
-    LISTENING = False
-    STOP_REQUESTED = False
+    # Initialize state, set LISTENING to true if state file exists,
+    # sets false if not.
+    def __init__(self, file: str):
+        self.file = file
+        self.LISTENING = True if os.path.isfile(file) else False
+        self.STOP_REQUESTED = False
+
+    # Method to update the listening status of the State class
+    # object. Checks for on/off and creates/deletes state file.
+    def update_listening_status(self, status: str) -> Optional[str]:
+        if status == "AUTOSCANON":
+            self.LISTENING = True
+            open(self.file, "a").close()
+            return "==Auto Scanning Activated=="
+        elif status == "AUTOSCANOFF":
+            self.LISTENING = False
+            try:
+                os.remove(self.file)
+            except OSError:
+                pass
+            finally:
+                return "==Auto Scanning Deactivated=="
+        else:
+            # Should be impossible to get her but it's here to
+            # satisfy mypy
+            return None
+
+    # Method to return the current state notification message
+    def get_listening_status_notice(self) -> Optional[str]:
```

Comment:
This could potentially just return `str`, and the final `else` clause could either `raise ValueError(<some relevant message>)` or `return ""` (same goes for the above function)

Reply:
Going with the `raise ValueError()` option here as returning an empty string shouldn't be necessary ever, as it stands, and we should always expect the `AUTOSCANON` or `AUTOSCANOFF` keywords. Also raising an error should make it easier to locate any potential bugs in later updates to the code.